client/extensions/woocommerce/store-sidebar/index.js

```diff
@@ -286,6 +289,12 @@ class StoreSidebar extends Component {
 				const storeCountry = get( storeLocation, 'country' );
 				showAllSidebarItems = isStoreManagementSupportedInCalypsoForCountry( storeCountry );
 			}
+
+			// Don't show sidebar items if store's removed & user is going to redirect
+			// to WooCommerce after installation
+			if ( isStoreRemoved && shouldRedirectAfterInstall ) {
```

Comment:
All of this logic mutating `showAllSidebarItems` is a bit confusing, imo you could consider refactoring this out into a helper function to make `showAllSidebarItems` immutable